source/agora/consensus/EnrollmentManager.d

```diff
@@ -316,9 +330,10 @@ public class EnrollmentManager
 
         // K, frozen UTXO hash
         this.data.utxo_key = frozen_utxo_hash;
+        this.enroll_key = frozen_utxo_hash;
```

Comment:
So I don't see why we need `enroll_key` ?

Reply:
It's necessary for selecting a `enrolled_height` from the validator set table. The `enrolled_height` is set in the middle of adding the block. The function that updates the `enrolled_height` is the `addValidatedBlock` which is modified in Henry's PR, or PR #504 . The code is as follows.

```
 foreach (enrollment; block.header.enrollments)
            if (!this.enroll_man.updateEnrolledHeight(enrollment.utxo_key,
                block.header.height))
                assert(0);
```